src/xma/src/xmaplugin/xmaplugin.cpp

```diff
@@ -31,32 +31,60 @@ using namespace std;
 
 #define XMAPLUGIN_MOD "xmapluginlib"
 
-//Sarab: TODO .. Assign buffr object fields.. bank etc..
-//Add new API for allocating device only buffer object.. which will not have host mappped buffer.. This could be used for zero copy plugins..
-//NULL data pointer in buffer obj implies it is device only buffer..
-//Remove read/write before SyncBO.. As plugin should manage that using host mapped data pointer..
+
+int32_t create_bo(xclDeviceHandle dev_handle, XmaBufferObj& b_obj, uint32_t size, uint32_t ddr_bank, 
```

Comment:
It looks like `XmaBufferObj& b_obj` is a return container structure.  
Instead of relying on the calling proc to initialize this data to a know state, you might wish to have this function at the very beginning set all the contents to zeros.

Reply:
Looked into it. For this particular case keeping it same. Not exposing more higher level internal data structures to this function.